app/services/news_services/tweet_trending_news_item.rb

```diff
@@ -0,0 +1,173 @@
+module NewsServices
+  class TweetTrendingNewsItem < Patterns::Service
+    include Rails.application.routes.url_helpers
+    LOG_PREFIX = '[NewsTweet]'
+    MAX_ACCEPTABLE_REPLICATION_LAG = ::ApplicationHelper::MAX_ACCEPTABLE_REPLICATION_LAG
+
+    # Test run will not send to Twitter or save to database, but will still notify on slack
+    def initialize(min_tweet_interval: 4.hours, max_tweets_per_day: 4, min_votes_required: 10, leading_text: "", max_coin_tag_count: 3, test_run: false, store_test_run: false)
+      @min_tweet_interval = min_tweet_interval
+      @max_tweets_per_day = max_tweets_per_day
+      @min_votes_required = min_votes_required
+      @leading_text = leading_text
+      @max_coin_tag_count = max_coin_tag_count
+      @test_run = test_run
+      @store_test_run = store_test_run
+
+      @slack_channel = '#' + ENV.fetch('SLACK_CHANNEL_TRENDING_NEWS').gsub('#', '')
+      @news_votes = nil
+      @news_item = nil
+    end
+
+    def call
+      unless can_tweet?
+        puts "#{LOG_PREFIX} Cannot tweet."
+        return
+      end
+
+      fetch_trendingest_news_item
+      unless should_tweet?
+        puts "#{LOG_PREFIX} Should not tweet ##{@news_item.try(:id)}: #{@news_item.try(:title)}."
+        return
+      end
+
+      response = send_news_item_tweet
+      if !@test_run
+        if response.present?
+          store_news_item_tweet(response)
+        else
+          error_message = "Failed to send tweet ##{@news_item.try(:id)}: #{@news_item.try(:title)}."
+          puts "#{LOG_PREFIX} #{error_message}"
+          slack_client.chat_postMessage(channel: @slack_channel, text: error_message, as_user: true)
+        end
+      else
+        if @store_test_run && response.present?
+          store_news_item_tweet(response)
+        else
+          puts "#{LOG_PREFIX} Test run; not storing tweet."
+        end
+      end
+
+      response
+    end
+
+    private
+
+    # Up to 280 characters; links count as 23 characters
+    # Include up to `max_coin_tag_count` coin tags
+    def build_tweet
+      link = build_url news_item_path(id: @news_item.id, slug: @news_item.slug)
+      coin_tags_sorted_by_rank = @news_item.coins.pluck(:ranking, :symbol)
+        .sort_by { |coin| coin[0] || Float::INFINITY }
```

Comment:
What's the `INFINITY` for?

Reply:
It's possible the rank could be nil. Want to make sure those get placed at the end.